src/views/templates/list/VirtualMachineTemplatesList.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import {
+  TEMPLATE_TYPE_BASE,
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_TYPE_VM,
+} from 'src/views/catalog/templatescatalog/utils/constants';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  K8sResourceCommon,
+  ListPageBody,
+  ListPageHeader,
+  useK8sWatchResource,
+  useListPageFilter,
+  VirtualizedTable,
+} from '@openshift-console/dynamic-plugin-sdk';
+
+import { default as Columns } from './VirtualMachineTemplatesColumns';
+import { default as Row } from './VirtualMachineTemplatesRow';
+
+type VirtualMachineTemplateTableProps = {
+  kind: string;
+  data: K8sResourceCommon[];
+  unfilteredData: K8sResourceCommon[];
+  loaded: boolean;
+  loadError: any;
+};
+
+const VirtualMachineTemplateTable: React.FC<VirtualMachineTemplateTableProps> = ({
+  data,
+  unfilteredData,
+  loaded,
+  loadError,
+  kind,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  return (
+    <VirtualizedTable<K8sResourceCommon>
+      data={data}
+      unfilteredData={unfilteredData}
+      loaded={loaded}
+      loadError={loadError}
+      columns={Columns(t)}
+      Row={Row}
+      rowData={{ kind }}
+    />
+  );
+};
+
+const VirtualMachineTemplatesList = () => {
+  const { t } = useKubevirtTranslation();
+
+  const [templates, loaded, loadError] = useK8sWatchResource<V1Template[]>({
+    kind: 'Template',
+    selector: {
+      matchExpressions: [
+        {
+          operator: 'In',
+          key: TEMPLATE_TYPE_LABEL,
+          values: [TEMPLATE_TYPE_BASE, TEMPLATE_TYPE_VM],
```

Comment:
```suggestion
          operator: 'Exists',
          key: 'template.kubevirt.io/type',
```
the reason I like using `exist` is for currecness, we will support any template type, not just `base` and `vm`
the reason I like clear text instead of const is because it's clearer to read and debug in case you want to know what your filtering by, you don't need to look up the value of the const.

Reply:
We will support any template type, not just base and vm, but I was told that for now, we want to display only the VM Templates... so now I am confused. Seems that things have changed...

I would move those constants somewhere to some more generic place, for example _src/utils/constants.ts_, as those are already useful in more places. Copying the same strings representing the same info doesn't seem to me as the best for providing more reusable and understandable code. And looking up the value of the const takes me one click. IMO using the same constant in more places also provides the info about the fact that the string represents the same information, also shows the connections between different parts of the code.

Reply:
> We will support any template type, not just base and vm, but I was told that for now, we want to display only the VM Templates... so now I am confused. Seems that things have changed...

we started supporting any template since 4.10, the main reason for the template list redesign is to better align with the beck-end who is supporting common templates and non common templates from any namespace.

in cases when design is not clear ping @tnisan or me

Reply:
> I would move those constants somewhere to some more generic place,

once the plugin is stable and we can go over it and find commonly used patterns, we may choose some constants used all over the place, document them (so every one can re-use them safely and currectly) version them (so changes done for one components can be tracked in case they affect other components) and put them in the shared directory

until them, create constants inside the extension you are working on, we will have time to do optimizations and collect shared components and constants once the plugin get stable. 

the lable selector currently used by the template list should not be shared by other extinctions, you do not want an implementation  details in the templates list to affect the virtualmachine extension or the catalog, they may have other needs, even if currently they may use the same label, it does not mean they will continue to use the same value.
you will end up with two similar const one for template extension and one for the other extension, or worse a bug in one of them if the const is changed to fix one of the extensions.

lets say the template list use TEMPLATE_TYPE_LABEL and the catalog use the same label, but in some future we have a bug and change the value of TEMPLATE_TYPE_LABEL for the list view, suddenly the catalog will fail, and we will not know why, it uses TEMPLATE_TYPE_LABEL, it will not be trivial to understand the the value was changed to fix a bug in the template list,

Reply:
> Catalog contains the same templates

no, we don't know that now, catalog may have other needs, the design of the catalog is not finished, and we certainly don't know what requirements we will get in the future.

once we have a stable code we can look for common patterns, and do other optimizations, until then, please only work within the code space of the extension you are working on.

as @vojtechszocs suggested we will have to have clear guidelines how to identify common patterns, how many times a common code should appear in the code, and size the repeatable code should be, what is ok to repeat in the name of readability and correctness, until we have stable code we can look for patterns in, and until we agree on what is a repeatable pattern, lets be safe and use the extension space.